### PR TITLE
Refactor Default Benchmark

### LINKED ISSUE #5143: Make algorithm benchmark per brokerage model
#### Expected Behavior

- Default benchmark does not cause live trading issues
#### Actual Behavior

- Since the default benchmark is SPY it causes live trading issue when trading with crypto brokerages or generally brokerages for which the user doesn't have a SPY subscription (like IB)
#### Potential Solution

- Make default algorithm benchmark part of the brokerage model. Implement accordingly for the different models
#### Reproducing the Problem

N/A
#### System Information

N/A
#### Checklist

- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by C-SELLERS**
#### Description

Have Brokerage models set the default benchmark for their brokerage.

#### Related Issue

Closes #5143 

#### Motivation and Context

> Since the default benchmark is SPY it causes live trading issue when trading with crypto brokerages or generally brokerages for which the user doesn't have a SPY subscription (like IB)

Instead of using a default benchmark for all brokerages this allows a unique benchmark to be used by any brokerage implementation through overriding `GetBenchmark()` in any `IBrokerageModel`. All it needs to do is return a `IBenchmark` and it will be set for the algorithm. 

Any brokerage model that does not override the function will take on the default benchmark (SPY).

#### How Has This Been Tested?

All regressions pass as expected.
Added a new regression to test the case of 0 value benchmark for C# and Python

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Review by C-SELLERS — COMMENTED** [review]
This works as it is right now but it is not smooth yet,  

An alternative option I am considering is having the the creation of the `IBenchmark` handled inside of the `IBrokerageModel` at a function like `GetBenchmark` so that it can be overridden by its inheritors.

Then we can have the case where if `_benchmarkSymbol == null` we can just use `Benchmark = BrokerageModel.GetBenchmark`

then the else is using the already defined `_benchmarkSymbol` as it is right now, creating the symbol, security, and then function.

**Review by jaredbroad — APPROVED** [review]

**Comment by C-SELLERS** [comment]
Sorry for two force pushes, had to rebase twice. The first one to get the new csproj format and then the second because master was broken from a different issue after the first time.

Its all fixed now though so no more rebases are necessary.

**Review by C-SELLERS — COMMENTED** [review]
Ready for review 👍🏽

File: Algorithm/QCAlgorithm.cs [hunk 1]
```diff
@@ -1106,7 +1087,15 @@ public void SetBenchmark(Symbol symbol)
...
-            _benchmarkSymbol = symbol;
+
+            // Create a security from this symbol
+            var security = Securities.CreateSecurity(symbol,
+                new List<SubscriptionDataConfig>(),
+                leverage: 1,
+                addToSymbolCache: false);
+
+            // Create and set our SecurityBenchmark
+            Benchmark = new SecurityBenchmark(security);
```
Comment (Martin-Molinero) [inline Algorithm/QCAlgorithm.cs:1098]: Nice! 👍

**Review by Martin-Molinero — APPROVED** [review]
Looks good minor findings 👍

**Review by C-SELLERS — COMMENTED** [review]
Review addressed 👍🏽 Thinking I might make a Python regression that matches the C# one for the sake of testing the wrapper.

**Review by Martin-Molinero — APPROVED** [review]
Nice work! 💯

**Pull request merged**